drivers/dma/dma_mcux_lpc.c

```diff
@@ -188,10 +194,66 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	if (!config->source_chaining_en && !config->dest_chaining_en) {
+		/* IFF channel is connected to OTrig Mux Then Disconnect */
+		if (!k_sem_take(&is_otrig_being_configured, K_MSEC(200))) {
+			for (int i = 0; i <  dev_config->num_of_otrigs; i++) {
+				if (dma_data->linked_otrig_ch[i] != -1 &&
+				   (config->linked_channel == dma_data->linked_otrig_ch[i] ||
+				    channel == dma_data->linked_otrig_ch[i])) {
+					dma_data->linked_otrig_ch[i] = -1;
+					break;
+				}
+			}
+			k_sem_give(&is_otrig_being_configured);
+		} else { /* k_sem_take(..); */
+			LOG_ERR("DMA attempted to configure an Otrig mux"
+				"while the mux was in use.");
+			return -EACCES;
+		}
+	} else {
+		if (!k_sem_take(&is_otrig_being_configured, K_MSEC(200))) {
+			LOG_DBG("link dma 0 channel %d with channel %d",
+				channel, config->linked_channel);
+			uint8_t is_otrig_available = 0;
+			uint8_t otrig_index = 0;
+
+			for (; otrig_index < dev_config->num_of_otrigs; ++otrig_index) {
```

Comment:
reminding to change this because github marked the other comment as outdated and hid it

Reply:
Fixed.